### PR TITLE
Fix OSX build

### PR DESCRIPTION
**Pull request by m-seker**
**Summary**

As the title says.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [x] Build-related changes
- [ ] Other, please describe:

If changing the UI of web configuration, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing setups:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's body (e.g. `Fixes: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated (docs/docs/en)
- [ ] Related tests have been updated

**PLEASE DON'T FORGET TO ADD YOUR CHANGES TO CHANGELOG.MD**
- [ ] Yes, CHANGELOG.md is also updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,7 +68,11 @@ ELSE()
 ENDIF()
 
 if ( NOT DEFINED PLATFORM )
-	if ( "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86" )
```
Comment (m-seker) [inline CMakeLists.txt:71]: This matches apple too (x86_64)

**Comment by tpmodding** [comment]
`➜  build git:(master) cmake -DPLATFORM=osx -DCMAKE_BUILD_TYPE=Debug ..
-- CMake Version: 3.18.1
CMake Deprecation Warning at CMakeLists.txt:18 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- PLATFORM: osx
-- ENABLE_AMLOGIC = OFF
-- ENABLE_DISPMANX = OFF
-- ENABLE_FB = OFF
-- ENABLE_OSX = OFF
-- ENABLE_SPIDEV   = OFF
-- ENABLE_TINKERFORGE = OFF
-- ENABLE_V4L2 = OFF
-- ENABLE_WS281XPWM = OFF
-- ENABLE_AVAHI = ON
-- ENABLE_USB_HID = OFF
-- ENABLE_CEC = OFF
-- ENABLE_X11 = ON
-- ENABLE_XCB = ON
-- ENABLE_QT = ON
-- ENABLE_TESTS = OFF
-- ENABLE_PROFILER = OFF
-- ENABLE_EXPERIMENTAL = OFF
-- validate json file
-- Current Version: 2.0.0-alpha.7
--  - Build: master (GitHub-4b44e7a0/4099d12d-1596898283)
-- GNU gold linker is not available, falling back to default system linker
-- Found Qt Version: 5.15.0
-- Found libusb-1.0:
--  - Includes: /usr/local/include/libusb-1.0
--  - Libraries: /usr/local/lib/libusb-1.0.dylib
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
CMake Error at dependencies/CMakeLists.txt:31 (add_subdirectory):
  The source directory

    /Users/Serhan/Downloads/hyperion/osx_build/dependencies/external/flatbuffers

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:39 (export):
  export given target "flatc" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:48 (get_property):
  get_property could not find TARGET flatc.  Perhaps it has not yet been
  created.


-- Using flatbuffers compiler: 
CMake Error at dependencies/CMakeLists.txt:99 (add_subdirectory):
  The source directory

    /Users/Serhan/Downloads/hyperion/osx_build/dependencies/external/protobuf

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:107 (export):
  export given target "protoc_compiler" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:115 (get_property):
  get_property could not find TARGET protoc_compiler.  Perhaps it has not yet
  been created.


-- Using protobuf compiler: 
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.8/bin/python3.8 (found version "3.8.5") found components: Interpreter 
-- Using static mbedtls libraries (build version "2.23.0")
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.8/bin/python3.8 (found version "3.8.5") found components: Interpreter Development Development.Module Development.Embed 
CMake Error at /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find X11 (missing: X11_X11_INCLUDE_PATH X11_X11_LIB)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindX11.cmake:407 (find_package_handle_standard_args)
  libsrc/grabber/x11/CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/Serhan/Downloads/hyperion/osx_build/build/CMakeFiles/CMakeOutput.log".`

**Comment by tpmodding** [comment]
➜  build git:(master) cmake -DPLATFORM=osx -DCMAKE_BUILD_TYPE=Debug ..  
-- CMake Version: 3.18.1
CMake Deprecation Warning at CMakeLists.txt:18 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- PLATFORM: osx
-- ENABLE_AMLOGIC = OFF
-- ENABLE_DISPMANX = OFF
-- ENABLE_FB = OFF
-- ENABLE_OSX = OFF
-- ENABLE_SPIDEV   = OFF
-- ENABLE_TINKERFORGE = OFF
-- ENABLE_V4L2 = OFF
-- ENABLE_WS281XPWM = OFF
-- ENABLE_AVAHI = ON
-- ENABLE_USB_HID = OFF
-- ENABLE_CEC = OFF
-- ENABLE_X11 = ON
-- ENABLE_XCB = ON
-- ENABLE_QT = ON
-- ENABLE_TESTS = OFF
-- ENABLE_PROFILER = OFF
-- ENABLE_EXPERIMENTAL = OFF
-- validate json file
-- Current Version: 2.0.0-alpha.7
--  - Build: master (GitHub-4b44e7a0/4099d12d-1596898283)
-- GNU gold linker is not available, falling back to default system linker
-- Found Qt Version: 5.15.0
-- Found libusb-1.0:
--  - Includes: /usr/local/include/libusb-1.0
--  - Libraries: /usr/local/lib/libusb-1.0.dylib
-- Looking for strtof_l
-- Looking for strtof_l - not found
-- Looking for strtoull_l
-- Looking for strtoull_l - not found
Proceeding with version: 1.12.0.103
-- Using flatbuffers compiler: /Users/Serhan/Downloads/hyperion/osx_build/build/bin/flatc
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Protobuf: doing try-compiles for hash map/set headers
-- Protobuf: doing try-compiles for pthread test
-- Looking for stddef.h
-- Looking for stddef.h - found
CMake Deprecation Warning at dependencies/external/protobuf/src/CMakeLists.txt:329 (cmake_policy):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Using protobuf compiler: /Users/Serhan/Downloads/hyperion/osx_build/build/bin/protoc
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.8/bin/python3.8 (found version "3.8.5") found components: Interpreter 
-- Using static mbedtls libraries (build version "2.23.0")
-- Found Python3: /usr/local/Frameworks/Python.framework/Versions/3.8/bin/python3.8 (found version "3.8.5") found components: Interpreter Development Development.Module Development.Embed 
CMake Error at /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find X11 (missing: X11_X11_INCLUDE_PATH X11_X11_LIB)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.18.1/share/cmake/Modules/FindX11.cmake:407 (find_package_handle_standard_args)
  libsrc/grabber/x11/CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/Serhan/Downloads/hyperion/osx_build/build/CMakeFiles/CMakeOutput.log".
See also "/Users/Serhan/Downloads/hyperion/osx_build/build/CMakeFiles/CMakeError.log".

**Comment by m-seker** [comment]
Looks like something like this is missing :
````
if (APPLE OR "${PLATFORM}" MATCHES "osx")
    set(ENABLE_OSX ON)
endif()
````

But your error happens because of something else ?

**Comment by tpmodding** [comment]
https://github.com/hyperion-project/hyperion.ng/blob/4099d12d9a8680ef0c89824d9f9af0a2f3b0a5e9/CMakeLists.txt#L115

this line should trigger it on...or?
but yep, the failure is here
  Could NOT find X11 (missing: X11_X11_INCLUDE_PATH X11_X11_LIB)

**Comment by m-seker** [comment]
You are right, interesting.

**Pull request merged**